app/celery/research_mode_tasks.py

```diff
@@ -114,43 +175,15 @@ def firetext_callback(notification_id, to):
     return {"mobile": to, "status": status, "time": "2016-03-10 14:17:00", "reference": notification_id}
 
 
-@notify_celery.task(bind=True, name="create-fake-letter-response-file", max_retries=5, default_retry_delay=300)
-def create_fake_letter_response_file(self, reference):
-    now = datetime.utcnow()
-    dvla_response_data = f"{reference}|Sent|0|Sorted|{now.date().isoformat()}"
-
-    # try and find a filename that hasn't been taken yet - from a random time within the last 30 seconds
-    for i in sorted(range(30), key=lambda _: random.random()):
-        upload_file_name = "NOTIFY-{}-RSP.TXT".format((now - timedelta(seconds=i)).strftime("%Y%m%d%H%M%S"))
-        if not file_exists(current_app.config["S3_BUCKET_DVLA_RESPONSE"], upload_file_name):
-            break
-    else:
-        raise ValueError(
-            f"cant create fake letter response file for {reference} - too many files for that time already exist on s3"
-        )
-
-    s3upload(
-        filedata=dvla_response_data,
-        region=current_app.config["AWS_REGION"],
-        bucket_name=current_app.config["S3_BUCKET_DVLA_RESPONSE"],
-        file_location=upload_file_name,
-    )
-    current_app.logger.info(
-        "Fake DVLA response file %s, content [%s], uploaded to %s, created at %s",
-        upload_file_name,
-        dvla_response_data,
-        current_app.config["S3_BUCKET_DVLA_RESPONSE"],
-        now,
-    )
-
-    # on development we can't trigger SNS callbacks so we need to manually hit the DVLA callback endpoint
-    if current_app.config["NOTIFY_ENVIRONMENT"] == "development":
-        make_request("letter", "dvla", _fake_sns_s3_callback(upload_file_name), None)
-
-
-def _fake_sns_s3_callback(filename):
-    message_contents = '{"Records":[{"s3":{"object":{"key":"%s"}}}]}' % (filename)  # noqa
-    return json.dumps({"Type": "Notification", "MessageId": "some-message-id", "Message": message_contents})
+@notify_celery.task(bind=True, name="create-fake-letter-callback", max_retries=3, default_retry_delay=60)
+def create_fake_letter_callback(self, notification_id: uuid.UUID, billable_units: int, postage: str):
```

Comment:
unfortunately i think we need to create this celery task in a separate PR first, or during the deploy the celery workers running the old code will error if they pick up one of these tasks without having a function to call for it

Reply:
I've split the code which adds the new task into [a separate PR](https://github.com/alphagov/notifications-api/pull/4303), and will rebase this PR after the new code has been added

Reply:
The code related to adding and using the new task for test letters has been added in https://github.com/alphagov/notifications-api/pull/4303 and https://github.com/alphagov/notifications-api/pull/4306, so this PR now solely deals with removing code.